src/tests/createHeadlessForm.test.js

```diff
@@ -1008,6 +1008,48 @@ describe('createHeadlessForm', () => {
         ]);
       });
 
+      it('supports nested group arrays', () => {
+        const result = createHeadlessForm({
+          type: 'object',
+          required: ['steps'],
+          properties: {
+            steps: {
+              type: 'array',
+              required: ['ingredients'],
+              additionalItems: false,
+              'x-jsf-presentation': {
+                inputType: 'group-array',
+                addFieldText: 'Add new step',
+              },
+              items: {
```

Comment:
it's true Jakob, this is something that we didn't build because we don't have nested group-arrays at Remote.

Reply:
It's not just groups, it's any kind of array with children that are not objects, so `string[]` or `number[]` don't work either.

Reply:
I see. Visually speaking, every form field must be an `object` to declare at least the `title` (label), that's why it's made to support just `object[]`. In the same way that the `jsonSchema` passed to `createHeadlessForm()` must also be an object with `properties`, otherwise it will throw.

Reply:
```
{
  "type": "object",
  "required": ["steps"],
  "title" : "Steps to reproduce the bug",
  "description": "What did you do to cause this?",
  "properties": {
    "steps": {
      "type": "array",
      "x-jsf-presentation": {
        "inputType": "group-array",
        "addFieldText": "Add new step"
      },
      "items": {
        "title": "Step",
        "type": "string"
      }
    }
  }
}
```

## Could result in:

---

### Steps to reproduce the bug

What did you do to cause this?

1. Step
  `Open app`
2. Step
  `Go to foo page`
3. Step
  `Press bar button|`

<kbd>+ Add new step</kbd>

<kbd>Submit</kbd>